cmd/build.go

```diff
@@ -158,17 +158,46 @@ func runBuild(cmd *cobra.Command, _ []string, newClient ClientFactory) (err erro
 		cfg buildConfig
 		f   fn.Function
 	)
-	if cfg, err = newBuildConfig().Prompt(); err != nil { // gather values into a single instruction set
+
+	// Initialize config first
+	cfg = newBuildConfig()
+
+	// Create function object to check if initialized
+	if f, err = fn.NewFunction(cfg.Path); err != nil {
 		return
 	}
-	if err = cfg.Validate(); err != nil { // Perform any pre-validation
-		return
+
+	// Check if function exists BEFORE prompting for config
+	if !f.Initialized() {
+		return fmt.Errorf(`no function found in current directory (or --path not specified).
+You need to point to a function directory to build.
```

Comment:
Im explicitly favoring the "current directory" approach and saying there is an optional `--path` argument after since most people will probably use the current directory.
```suggestion
		return fmt.Errorf(`no function found in provided path (current directory or via --path).
You need to be in a function directory (or use --path)
```

Reply:
so i guess we should close this pr !